backend/models/userModel.js

```diff
@@ -1,67 +1,71 @@
 // connect to mongo
 const mongoose = require('mongoose');
-const { v4: uuid } = require('uuid');
 const bcrypt = require('bcrypt');
 const generateAuthToken = require('../utils/tokenUtil');
 
 // Define the user schema
 const userSchema = new mongoose.Schema({
-    id: { type: String, required: true, unique: true },
-    creation_date: { type: Date, required: true },
-    auth_token: { type: String },
-    email: { type: String },
-    first_name: { type: String, required: true },
-    last_name: { type: String, required: true },
-    username: { type: String },
-    password: { type: String },
-    avatar: { type: String }, // Assuming the avatar is stored as a file path or URL
-    decks: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Deck' }],
+  creationDate: {type: Date, required: true},
+  authToken: {type: String, default: null},
+  email: {type: String},
+  firstName: {type: String, required: true},
+  lastName: {type: String, required: true},
+  username: {type: String},
+  password: {type: String},
+  avatar: {type: String}, // Assuming the avatar is stored as a file path or URL
+  decks: [{type: mongoose.Schema.Types.ObjectId, ref: 'Deck'}],
 });
 
-userSchema.statics.createUser = async function (first_name, last_name, email, password) {
-    try {
+userSchema.statics.createUser =
+    async function(firstName, lastName, email, password) {
+      try {
         const newUser = new this({
-            id: uuid(),
-            creation_date: new Date(),
-            email,
-            first_name,
-            last_name,
-            password: await bcrypt.hash(password, 10),
-            decks: [],
+          creationDate: new Date(),
+          email,
+          firstName,
+          lastName,
+          password: await bcrypt.hash(password, 10),
+          decks: [],
         });
 
         const user = await newUser.save();
 
-        const token = generateAuthToken();
-        user.auth_token = token;
         await user.save();
```

Comment:
why are u deleting the auth token for a new user?

Reply:
not necessary for when we creating a user